### ISSUE TITLE
[Bug] docsearch fails to display page titles in vuepress-theme-hope

### ISSUE THREAD

**Issue by rockbenben** [vuepress-theme-hope/vuepress-theme-hope#4470]
### Checklist

- [X] I am using **LTS** version of Node.js.
- [X] I am using the **latest v2 version** of VuePress and VuePress official plugins.
- [X] I am using the **latest v2 version** of vuepress-theme-hope and all plugins in this repo.
- [X] I followed the docs and I double checked my configuration.
- [X] I have searched [FAQ](https://theme-hope.vuejs.press/faq/) and I did not find the answer.

### Package name

vuepress-theme-hope

### Which operating system are you using?

- [ ] macOS
- [X] Windows
- [ ] Linux

### Describe the bug

I am experiencing an issue where `@vuepress/plugin-docsearch` is not displaying page titles correctly when used with the `vuepress-theme-hope`. I tested the same search on two different VuePress-based sites, and the results were inconsistent.

### Steps to Reproduce
1. On the [VuePress Ecosystem website](https://ecosystem.vuejs.press/zh/plugins/search/docsearch.html), I performed a search for a specific keyword using `docsearch`.
2. On the [vuepress-theme-hope website](https://theme-hope.vuejs.press/zh/guide/feature/search.html), I searched for the same keyword using `docsearch`.

**Expected Results**:  
The search results should display the page titles correctly in both instances.

**Actual Results**:  
- On the VuePress Ecosystem site, the page titles are displayed correctly in the search results.
![image](https://github.com/user-attachments/assets/b704fc3b-a335-4b95-9286-9a21a2dea88b)
- On the `vuepress-theme-hope` site, the page titles are missing from the search results.
![image](https://github.com/user-attachments/assets/9c7f7bbc-479d-45a5-94fb-1ca9dffe4297)

### Additional context

I have verified that the page titles are being correctly indexed by Algolia. As shown in the attached screenshot of the Algolia index, `lvl0` captures the page titles correctly, but these are not appearing in the search results on the `vuepress-theme-hope` site.
![image](https://github.com/user-attachments/assets/7704a50c-9c39-4e39-9cb4-1cb46e19ce80)

**Comment by Mister-Hope** [comment]
![image](https://github.com/user-attachments/assets/41df4a38-2c9b-4277-bfc2-55f9fcde21d0)

My theme does not make any customization on the docsearch plugin, so this should probably be a crawler thing.

**Comment by Mister-Hope** [comment]
I do think that this is because the ecosystem site has an incorrect config that has an outdated sidebar.

**Comment by Mister-Hope** [comment]
It's a little weird for me as I already unify their crawler config and recrawl them but their behaviors are still different.

**Comment by rockbenben** [comment]
This issue occurs not only on my site but also in the Algolia search preview in their editor. 

I compared my configuration with the [VuePress Ecosystem site](https://ecosystem.vuejs.press/zh/plugins/search/docsearch.html), and the only significant difference I found was in the page structure. Could this issue be related to an incomplete recommendation in the Ecosystem configuration?

```yml
      recordExtractor: ({ $, helpers }) => {
        return helpers.docsearch({
          recordProps: {
            lvl0: {
              selectors: [".vp-sidebar-page.active", ".vp-page-title h1"],
              defaultValue: "Documentation",
            },
            // vuepress-theme-hope 默认的容器类名为 theme-hope-content
            lvl1: ".theme-hope-content h1",
            lvl2: ".theme-hope-content h2",
            lvl3: ".theme-hope-content h3",
            lvl4: ".theme-hope-content h4",
            lvl5: ".theme-hope-content h5",
            lvl6: ".theme-hope-content h6",
            content: ".theme-hope-content p, .theme-hope-content li",
          },
          indexHeadings: true,
        });
      },
```

![image](https://github.com/user-attachments/assets/5920fe3e-c12f-4d79-9e26-d1e902aba4e5)

**Comment by Mister-Hope** [comment]
any progress with this?

Maybe you can try support from algolia.

**Comment by rockbenben** [comment]
For your information, upon usage today, the issue was found to be resolved, with the current version being 2.0.0-rc.58.

![image](https://github.com/user-attachments/assets/1f0786a8-a97e-4657-9014-59cb0b0325f0)
